openverse_catalog/dags/providers/provider_api_scripts/wikimedia_commons.py

```diff
@@ -21,450 +21,424 @@
+
+        record_data = {
+            "media_url": media_url,
+            "foreign_landing_url": media_info.get("descriptionshorturl"),
+            "foreign_identifier": foreign_id,
+            "license_info": license_info,
+            "creator": creator,
+            "creator_url": creator_url,
+            "title": title,
+            "filetype": filetype,
+            "filesize": filesize,
+            "meta_data": meta_data,
+            "media_type": valid_media_type,
+        }
+
+        # Extend record_data with media-type specific fields
+        funcs = {
+            IMAGE: self.get_image_record_data,
+            AUDIO: self.get_audio_record_data,
+        }
+        return funcs[valid_media_type](record_data, media_info)
+
+    def get_image_record_data(self, record_data, media_info):
+        """Extend record_data with image-specific fields"""
+        record_data["image_url"] = record_data.pop("media_url")
+        if record_data["filetype"] == "svg":
+            record_data["category"] = "illustration"
+
+        return {
+            **record_data,
+            "width": media_info.get("width"),
+            "height": media_info.get("height"),
+        }
+
+    def get_audio_record_data(self, record_data, media_info):
+        """Extend record_data with audio-specific fields"""
+        record_data["audio_url"] = record_data.pop("media_url")
+
+        duration = int(float(media_info.get("duration", 0)) * 1000)
+        record_data["duration"] = duration
+        record_data["category"] = self.extract_audio_category(record_data)
+
+        file_metadata = self.parse_audio_file_meta_data(media_info)
+        if sample_rate := self.get_value_by_names(
+            file_metadata, ["audio_sample_rate", "sample_rate"]
+        ):
+            record_data["sample_rate"] = sample_rate
+        if bit_rate := self.get_value_by_names(
+            file_metadata, ["bitrate_nominal", "bitrate"]
+        ):
+            record_data["bit_rate"] = bit_rate if bit_rate <= 2147483647 else None
+        if channels := self.get_value_by_names(
+            file_metadata, ["audio_channels", "channels"]
+        ):
+            record_data["meta_data"]["channels"] = channels
+
+        return record_data
+
+    def parse_audio_file_meta_data(self, media_info):
+        """Parse out audio file metadata"""
+        metadata = media_info.get("metadata", [])
+        streams = self.get_value_by_name(metadata, "streams")
+        try:
+            if not streams:
+                audio = self.get_value_by_name(metadata, "audio")
+                streams = self.get_value_by_name(audio, "streams")
+            streams_data = [stream["value"] for stream in streams][0]
+            file_data = self.get_value_by_name(streams_data, "header")
+            if not file_data:
+                file_data = streams_data
+        except (IndexError, TypeError):
+            file_data = []
+        return file_data
```

Comment:
I know this is unaltered from the OG script, but could we alter this so that it checks for keys/indices rather than a try/except block?

Reply:
This bit of logic is actually somewhat fiddly because of the structure of the data we get back from the API, which involves lists of dicts with "name" and "value" keys ([example file](https://github.com/WordPress/openverse-catalog/blob/main/tests/dags/providers/provider_api_scripts/resources/wikimedia/audio_filedata_ogg.json#L7)). I pushed a refactor to remove the try/catch safely but I can see why they were added 😄